framework/src/play/classloading/ApplicationCompiler.java

```diff
@@ -37,14 +37,13 @@
     ApplicationClasses applicationClasses;
     Map<String, String> settings;
     private static final String JAVA_SOURCE_DEFAULT_VERSION = "1.8";
-    static final Map<String, String> compatibleJavaVersions = new HashMap<>();
-    
-    static {
-    	compatibleJavaVersions.put("1.8", CompilerOptions.VERSION_1_8);
-    	compatibleJavaVersions.put("9", CompilerOptions.VERSION_9);
-    	compatibleJavaVersions.put("10", CompilerOptions.VERSION_10);
-    	compatibleJavaVersions.put("11", CompilerOptions.VERSION_11);
-	}
+    static final Map<String, String> compatibleJavaVersions = Map.of( //
+    		"1.8", CompilerOptions.VERSION_1_8, //
+    		"9", CompilerOptions.VERSION_9, //
+    		"10", CompilerOptions.VERSION_10, //
+    		"11", CompilerOptions.VERSION_11, //
+    		"12", CompilerOptions.VERSION_12 //
```

Comment:
Shouldn't we add "13" and "14" versions?

Reply:
Option "13" is not available with this version of the JDT compiler (see previous remarks).
Option "14" is not available even in the latest version of the lib.

Reply:
it is already dine in #1338